### DISCUSSION TITLE
Know how question: what your experience with cam reflections inside the dome / pictures?

### DISCUSSION THREAD

**Discussion by jkoenig72** [aaronwmorris/indi-allsky#764] [General]
Just wondering if someone has found a clever solution for this. I see quite clear cam reflections in my allsky cams. 

It looks to me that it's worse if the cam lens is exposed more "out" of the case to the dome if the cam lens is relatively flat with the case, it does not seem that bad. 

Maybe someone found a good and simple way to minimize reflections.  

Thanks!
Joerg

**Comment by aaronwmorris** [comment]
I have pretty bad reflections, as well.  The new stretching algorithm exacerbates the problem by really bringing out the reflections.  

I think the only real fix to this is a black, matte paint that is designed to be in the sun.

**Comment by bluthen** [comment]
I do the opposite, I try to put the camera/lens as close to the top of the dome as I can. Go through less of the dome and get less distortion and better focus. But I'm probably using different materials, lenses, and have different points of reflection. We got a different setup, so to each their own to experiment with.

**Comment by aaronwmorris** [comment]
I think there was a post in the Facebook Allsky Camera group where someone experimented with different heights of the camera lens inside the dome.

The dome is effectively acting as a concave mirror.  If you can orient the camera lens at the focal point of the dome reflections, I think it might eliminate the reflections (somewhat).

Kind of like looking at your reflection in a spoon.  The focal point is where the image switches between upright and inverted.

**Reply by SubBass100** [reply]
I make sure to use glass. I also keep the lens really close to the glass (without touching). Lastly I use rainx spray on the inside and outside of the glass, keeps the water and moisture off. I designed a 3d printed wall mount that holds a flower centerpiece (9 bucks) it works really well. I'll post to "showandtell" when I get images of it.

**Reply by Vertikar** [reply]
> I make sure to use glass. I also keep the lens really close to the glass (without touching). Lastly I use rainx spray on the inside and outside of the glass, keeps the water and moisture off. I designed a 3d printed wall mount that holds a flower centerpiece (9 bucks) it works really well. I'll post to "showandtell" when I get images of it.

What glass dome are you using?

**Reply by StarGeezerPhil** [reply]
@SubBass100 also interested to know details of your glass dome - I've looked but had no success finding any.

Sorry - see you answered in a separate reply.

**Comment by SubBass100** [comment]
I use this, It's cheap and clear. Works really well. https://www.amazon.com/gp/product/B09PHBP19F/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1

I posted the project to "Show and Tell" https://github.com/aaronwmorris/indi-allsky/discussions/949

**Comment by StarGeezerPhil** [comment]
I didn't do rigorous testing, but during building the enclosure it appeared that the reflections were minimised when my lens was closer to the dome. 

Currently using a cheap acryllic dome from ebay and the IMX378-190 waveshare Pi camera module.

To minimise reflections I've flocked around the camera and the enclosure within the dome itself, which has made a big improvement to the amount of reflections picked up.

**Comment by wjschmitt** [comment]
I fixed the problems with glare/reflections on the dome by just getting rid of the dome.  I also don't have any more problems with dew and no longer need a dew heater.  I have the camera inside my enclosure with just the lens sticking out the top.

**Reply by gigaklatch** [reply]
Perhaps your all-sky camera is not deployed 24/7 then? Or how will it withstand rain, snow and such? And birds..?

**Reply by Vertikar** [reply]
I was wondering the same thing...

**Reply by wjschmitt** [reply]
The camera has been out 24/7 for the past week, through several rain storms.  We don't get much snow here in Florida so I can't comment on that.  So far, no problems with birds. but I image it should handle them better since it is a smaller target than a dome.  The only thing exposed to the elements is the lens.  And if you look close, there is actually a small glass dome as the front element of the lens.  It costs only US$10 so it is easily replaced if necessary.

**Reply by gigaklatch** [reply]
But have you made any seals? Silicone or anything? How does it work if you need to refocus the lens? On my lens it moves the outer bit of the lens in or out. Perhaps you have a lens with internal focus mechanism?

Would you mind showing  some pictures?

**Reply by wjschmitt** [reply]
The top of the enclosure is covered in a 1mm silicone rubber sheet, which provides the seal when the lens is screwed into the camera body, which is inside the enclosure.  The lens is tightened to the body, but the HQ camera body has its own focusing ring that turns.  So I focus by rotating the entire enclosure top.  It is a bit tricky to get focus but hopefully I only have to do it once.

Here are some pics.  First is the M12 1.55mm lens with the hole in the enclosure lid.   Next is the lens in the hole.  This is my second case, so I haven't added the silicone sheet yet.  Then, this is the HQ camera body inside the lid with the lens screwed in.   Finally, a picture of my first version outside.  The ring around the lens is for shielding some of the lights from my house.

![lens_hole](https://github.com/aaronwmorris/indi-allsky/assets/118464192/6279079a-60ee-4f16-a77c-7347ec448899)
![lens_top](https://github.com/aaronwmorris/indi-allsky/assets/118464192/5a193f06-6d6f-4050-94fe-5dba11162994)
[
![HQ](https://github.com/aaronwmorris/indi-allsky/assets/118464192/83638fd8-4ed0-48f9-996e-fe3d73bd7b31)
](url)
![outside1](https://github.com/aaronwmorris/indi-allsky/assets/118464192/26c0a3f5-649c-4a5b-8d58-82fa01e9dbeb)
